src/Microsoft.Health.Operations.Functions/Management/PurgeOrchestrationInstanceHistory.cs

```diff
@@ -57,27 +55,39 @@ public async Task Run([TimerTrigger(PurgeFrequencyVariable)] TimerInfo myTimer,
         EnsureArg.IsNotNull(client, nameof(client));
         EnsureArg.IsNotNull(myTimer, nameof(myTimer));
         EnsureArg.IsNotNull(log, nameof(log));
+        IReadOnlyCollection<OrchestrationRuntimeStatus>? statuses = EnsureArg.HasItems(_options.Statuses, nameof(_options.Statuses));
+        TimeSpan minimumAge = TimeSpan.FromDays(EnsureArg.IsGt(_options.MinimumAgeDays, 0, nameof(_options.MinimumAgeDays)));
+        IReadOnlyCollection<string>? instancesToSkipPurging = _options.InstancesToSkipPurging == null ? Array.Empty<string>() : _options.InstancesToSkipPurging;
 
         log.LogInformation("Purging orchestration instance history at: {Timestamp}", Clock.UtcNow);
         if (myTimer.IsPastDue)
         {
             log.LogWarning("Current function invocation is running late.");
         }
 
-        DateTimeOffset end = Clock.UtcNow - _minimumAge;
-        log.LogInformation("Purging all orchestration instances with status in {{{Statuses}}} that started before '{End}'.",
-            string.Join(", ", _statuses),
-            end);
-
-        PurgeHistoryResult result = await client.PurgeInstanceHistoryAsync(DateTime.MinValue, end.UtcDateTime, _statuses);
+        DateTimeOffset end = Clock.UtcNow - minimumAge;
+        log.LogInformation("Purging all orchestration instances with status in {{{Statuses}}} that started before '{End}' and not in {{{ListofInstanceToSkipPurging}}}.",
+            string.Join(", ", statuses),
+            end,
+            string.Join(", ", instancesToSkipPurging));
 
-        if (result.InstancesDeleted > 0)
+        OrchestrationStatusQueryCondition condition = new OrchestrationStatusQueryCondition
         {
-            log.LogInformation("Deleted {Count} orchestration instances from storage.", result.InstancesDeleted);
-        }
-        else
+            CreatedTimeFrom = DateTime.MinValue,
+            CreatedTimeTo = end.UtcDateTime,
+            RuntimeStatus = statuses
+        };
+
+        var instances = await client.ListInstancesAsync(condition, CancellationToken.None);
+
+        var instancesToPurge = instances.DurableOrchestrationState.Where(x => !instancesToSkipPurging.Contains(x.Name, StringComparer.OrdinalIgnoreCase));
+
+        foreach(var instance in instancesToPurge)
         {
-            log.LogInformation("No Orchestration instances found within given parameters.");
+            var result = await client.PurgeInstanceHistoryAsync(instance.InstanceId);
+            log.LogInformation("Instance '{InstanceName}' deleted from storage", instance.Name);
         }
+
+        log.LogInformation("Deleted {Count} orchestration instances from storage.", instancesToPurge.Count());
```

Comment:
nit: technically this is agnostic of the back-end, so perhaps:
```suggestion
        log.LogInformation("Deleted {Count} orchestration instances from the task hub.", instancesToPurge.Count());
```